### PR TITLE
Add solution category to details page

### PR DESCRIPTION
**Pull request by bjss-colin**
Adds a new optional input to the Solution Details page. The implementation is largely complete but we're waiting for a description to add to the hint text for the new input.

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Shared/Details.cshtml [hunk 1]
```diff
@@ -36,24 +36,34 @@
...
-                                      data-test-id="framework-names" />
+                                      data-test-id="framework-names"/>
                     }
                 </nhs-checkbox-container>
             </nhs-fieldset-form-label>
             @for (var i = 0; i < Model.Frameworks.Count; i++)
             {
-                <input type="hidden" asp-for="@Model.Frameworks[i].Name" />
+                <input type="hidden" asp-for="@Model.Frameworks[i].Name"/>
             }
```
Comment (dssherlock) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Shared/Details.cshtml:47]: The question mark should technically come after the brackets

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Shared/Details.cshtml [hunk 2]
```diff
@@ -36,24 +36,34 @@
...
 
+            <nhs-fieldset-form-label label-text="What is the category of this solution? (optional)"
+                                     label-hint="This categorisation is assigned by NHS England and is based on the solution's capabilities, operational characteristics, and associated risks."
+                                     asp-for="SelectedCategory">
```
Comment (dssherlock) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Shared/Details.cshtml:50]: No way of de-selecting a Category once one has been chosen. Not ideal for an optional property?

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/AddCatalogueSolutionController.cs [hunk 3]
```diff
@@ -34,7 +34,7 @@ public async Task<IActionResult> Index()
             var model = new SolutionModel()
             {
                 BackLink = Url.Action(nameof(CatalogueSolutionsController.Index), typeof(CatalogueSolutionsController).ControllerName()),
```
Comment (dssherlock) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Controllers/AddCatalogueSolutionController.cs:36]: Why are we changing this content?

**Review by dssherlock — APPROVED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Shared/Details.cshtml (hunk 2 above)
Comment (ghost) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Shared/Details.cshtml:50]: It's optional until selected, then the working assumption is that any solution that has been categorised cannot be uncategorised.

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Shared/Details.cshtml (hunk 1 above)
Comment (ghost) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Shared/Details.cshtml:47]: I'll leave as is for now as this matches other pages on the Catalogue such as the Add/Edit Associated Service page.

**Pull request merged**